### ISSUE TITLE
Bug: `derive_param_tte()` fails if `dataset` and `by_vars` is specified

### ISSUE THREAD

**Issue by bundfussr** [pharmaverse/admiral#2336]
### What happened?

 `derive_param_tte()` fails if `dataset` and `by_vars` is specified because the check on duplicate parameter codes fails.

### Reproducible Example

```
library(dplyr)
library(lubridate)
library(admiral)
adsl <- tibble::tribble(
  ~USUBJID, ~TRTSDT,           ~TRTEDT,           ~EOSDT,
  "01",     ymd("2020-12-06"), ymd("2021-03-02"), ymd("2021-03-06"),
  "02",     ymd("2021-01-16"), ymd("2021-01-20"), ymd("2021-02-03")
) %>%
  mutate(STUDYID = "AB42")

ae <- tibble::tribble(
  ~USUBJID, ~AESTDTC,     ~AESEQ, ~AEDECOD,
  "01",     "2021-01-03",      1, "Flu",
  "01",     "2021-03-04",      2, "Cough",
  "01",     "2021-01-01",      3, "Flu"
) %>%
  mutate(
    STUDYID = "AB42",
    AESTDT = ymd(AESTDTC)
  )

ttae <- event_source(
  dataset_name = "ae",
  date = AESTDT,
  set_values_to = exprs(
    EVENTDESC = "AE",
    SRCDOM = "AE",
    SRCVAR = "AESTDTC",
    SRCSEQ = AESEQ
  )
)

eot <- censor_source(
  dataset_name = "adsl",
  date = pmin(TRTEDT + days(10), EOSDT),
  censor = 1,
  set_values_to = exprs(
    EVENTDESC = "END OF TRT",
    SRCDOM = "ADSL",
    SRCVAR = "TRTEDT"
  )
)

  derive_param_tte(
    dataset = adsl %>% select(STUDYID, USUBJID) %>% mutate(PARAMCD = "XYZ"),
    dataset_adsl = adsl,
    by_vars = exprs(AEDECOD),
    start_date = TRTSDT,
    event_conditions = list(ttae),
    censor_conditions = list(eot),
    source_datasets = list(adsl = adsl, ae = ae),
    set_values_to = exprs(
      PARAMCD = paste0("TTAE", as.numeric(as.factor(AEDECOD))),
      PARAM = paste("Time to First", AEDECOD, "Adverse Event"),
      PARCAT1 = "TTAE",
      PARCAT2 = AEDECOD
    )
  )
```
results in
```
Error in match(x, table, nomatch = 0L) : 
  'match' requires vector arguments
```

**Comment by bms63** [comment]
Is this a critical bug or one that we can wait on for a while?

**Comment by bundfussr** [comment]
As a work-around you could not specify the `dataset` argument, create separate datasets for each `derive_param_tte()` call, and bind them all together using `bind_rows()`.
So it is not super-critical. However, I am not sure if we should wait until June.

**Comment by bms63** [comment]
I will add to agenda for next week's meeting.

**Comment by millerg23** [comment]
@bms63 I looked at the Hotfix release page, do you create the `patch` branch that I should take feature branch from to fix this?

**Comment by bms63** [comment]
I think we should create a patch branch off the latest release tag and then branch off the patch to make the fix.  whew!!  @zdz2101

**Comment by millerg23** [comment]
OK thanks, I have created a branch called `patch` from tag `1.0.1` on GItHub

**Comment by bundfussr** [comment]
Our documentation is inconsistent. On the one hand it requires that the `patch` branch is protected and updated after each release. On the other hand it suggests to create the `patch` branch from a release tag.

And the formatting of the "Hot Fix Release" section got mixed up:
![image](https://github.com/pharmaverse/admiral/assets/80953585/fb0b2d7e-1f2f-44a2-ba75-d3029d681e32)

**Comment by bms63** [comment]
> Our documentation is inconsistent. On the one hand it requires that the `patch` branch is protected and updated after each release. On the other hand it suggests to create the `patch` branch from a release tag.
> 
> And the formatting of the "Hot Fix Release" section got mixed up: ![image](https://private-user-images.githubusercontent.com/80953585/307382134-fb0b2d7e-1f2f-44a2-ba75-d3029d681e32.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg3MjQ5NDcsIm5iZiI6MTcwODcyNDY0NywicGF0aCI6Ii84MDk1MzU4NS8zMDczODIxMzQtZmIwYjJkN2UtMWYyZi00NGEyLWJhNzUtZDMwMjlkNjgxZTMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIzVDIxNDQwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiNWRjNTJiY2M0ZDJlNmViODZlMGMxOGRhMjk0NjIzYzkzM2M0NDZiMGFhZWE4MTdhNGJiMTg0YmM1YzVlNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qcnrkghO-yaglvwiTHGMtiVLPO-NMONxqCjSxsc0vMg)

Thanks Stefan I created a new issue https://github.com/pharmaverse/admiraldev/issues/365
